IS1001Test.py

```diff
@@ -478,11 +478,23 @@ def _verify_error_response(self, test, response, status_code, error_value):
             self._raise_nmos_exception(
                 test, response, string="Incorrect Status Code Returned. Expected {}".format(status_code)
             )
-
-        if response.headers["Content-Type"] != "application/json":
-            raise NMOSTestException(test.FAIL(
-                "Body of Error was not JSON. Content-Type is '{}'".format(response.headers["Content-Type"])
-            ))
+        if "Content-Type" not in response.headers:
+            raise NMOSTestException(test.FAIL("Auth Server failed to signal its Content-Type."))
+        else:
+            ctype = response.headers["Content-Type"]
+            ctype_params = ctype.split(";")
+            if ctype_params[0] != "application/json":
+                raise NMOSTestException(test.FAIL(
+                    "Auth Server signalled a Content-Type of {} rather than application/json."
+                    .format(ctype)))
+            elif len(ctype_params) == 2 and ctype_params[1].strip().lower() == "charset=utf-8":

```

Comment:
I think this entire block of code could be pulled out and placed into a utils file like `NMOSUtils.py` to save us redefining it for each spec. It also doesn't catch the case when a different charset is used (which is more important than a redundant charset being present.)
```suggestion
            elif len(ctype_params) == 2 and ctype_params[1].strip().lower() != "charset=utf-8":
                    raise NMOSTestException(test.WARNING(
                    "Auth Server signalled an unsupported 'charset' in its Content-Type: {}"
                    .format(ctype)))
```

Reply:
`charset=foobar` is already caught by the following section and so reports a FAIL with "Auth Server signalled unexpected additional parameters in its Content-Type:".

`charset=utf-8` should be reported as a WARNING.

Reply:
I do agree it could be factored out, since already found in `IS0401Test.test_03`. PR #318 already proposes putting it in GenericTest (this is pure HTTP/IANA media-type, nothing NMOS specific, after all).

Reply:
Ah fair enough I missed the equals sign in the last conditional

Reply:
@dannymeloy You mention using this in `_verify_response` but I can't tell if that's only used with JSON responses, so whether additional logic would be required to cover redirects and other file types from the Auth API?

Reply:
There only appears to be one instance in which verify_response is called when its expecting a redirect (the others should all be JSON) so that should probably either be removed for consistency, or an extra parameter (and logic) added to the function (`... content_type='application/json'`) to allow for extra content types.